config/software/metasploit-framework.rb

```diff
@@ -95,4 +95,46 @@
     delete "#{install_dir}/devkit"
   end
   copy "#{project_dir}/Gemfile.lock", "#{install_dir}/embedded/framework/Gemfile.lock"
+
+  # Darwin needs extra tweaks to library files to work relatively.
+  #
+  # We need to replace all hard coded `/opt/metasploit-framework` load commands in library files, such as:
+  #
+  #    $ otool -L ./embedded/lib/ruby/gems/2.6.0/gems/eventmachine-1.2.7/lib/rubyeventmachine.bundle
+  #    ./embedded/lib/ruby/gems/2.6.0/gems/eventmachine-1.2.7/lib/rubyeventmachine.bundle:
+  #      @executable_path/../lib/libruby.2.6.dylib (compatibility version 2.6.0, current version 2.6.6)
+  #      /opt/metasploit-framework/embedded/lib/libssl.1.1.dylib (compatibility version 1.1.0, current version 1.1.0)
+  #      /opt/metasploit-framework/embedded/lib/libcrypto.1.1.dylib (compatibility version 1.1.0, current version 1.1.0)
+  #      /usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 120.1.0)
+  #      /usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1226.10.1)
+  #
+  # To instead use a relative path:
+  #      @executable_path/../lib/libcrypto.1.1.dylib
+  #
+  # Full context: https://github.com/rapid7/metasploit-omnibus/pull/127#issuecomment-632842474
+  if mac_os_x?
+    block do
+      build_files = Dir[
+        "#{install_dir}/embedded/bin/*",
+        "#{install_dir}/embedded/lib/**/*"
+      ]
+      absolute_lib_path = "#{install_dir}/embedded/lib"
+
+      build_files.each do |file|
```

Comment:
For these directory traversals, there can be shell scripts in addition to macho binaries. Ran into an error during the traversal from install_name_command where it couldn't patch a script:

> /Users/bcook/projects/metasploit-omnibus/config/software/metasploit-framework.rb:134:in `block (4 levels) in evaluate': Could not successfully patch load_command for relative dynamic linking for dependency '/opt/metasploit-framework/embedded/lib/postgresql/pgxs/config/install-sh' (RuntimeError)

Maybe this could employ something like the 'script_path?' function in metasploit-framework/lib/msf/core/modules/loader/base.rb to determine if something is a script before processing it.

Alternately, otool complains earlier which could also be used as a skip flag:

> /opt/metasploit-framework/embedded/lib/postgresql/pgxs/config/install-sh: is not an object file